go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1371,6 +1376,70 @@ func (call *builtinFromDays) compile(c *compiler) (ctype, error) {
 	return ctype{Type: sqltypes.Date, Flag: arg.Flag | flagNullable}, nil
 }
 
+func (b *builtinSecToTime) eval(env *ExpressionEnv) (eval, error) {
+	arg, err := b.arg1(env)
+	if arg == nil {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, err
+	}
+
+	var e *evalDecimal
+	prec := datetime.DefaultPrecision
+
+	switch {
+	case sqltypes.IsDecimal(arg.SQLType()):
+		e = arg.(*evalDecimal)
+	case sqltypes.IsIntegral(arg.SQLType()):
+		e = evalToDecimal(arg, 0, 0)
+	case sqltypes.IsTextOrBinary(arg.SQLType()):
+		b := arg.(*evalBytes)
+		if b.isHexOrBitLiteral() {
+			e = evalToDecimal(arg, 0, 0)
+		} else {
+			e = evalToDecimal(arg, 0, datetime.DefaultPrecision)
```

Comment:
Is this the right precision? Or should it be inferred from the input string in this case?

Reply:
I think this should be the correct precision for this case, as mysql seems to set 6 as precision for `*evalBytes`.